pkg/resource/generator.go

```diff
@@ -147,6 +147,12 @@ func (g *Generator) syncStorage(cr *imageregistryv1.Config) error {
 		if !exists {
 			runCreate = true
 		}
+
+		// GCP: If during previous reconcile error was set, failing to add required
+		// tags, setting runCreate to true  to retry adding the tags.
+		if cr.Spec.Storage.GCS != nil {
+			runCreate = true
+		}
```

Comment:
This feels hacky. Could you trigger recreation using the GCS driver's `StorageChanged` method? It's still kinda hacky but at least this generator wouldn't have to know about it.

Reply:
suggestion incorporated, moved the check to `StorageChanged` method.